src/btree/bt_split.c

```diff
@@ -1420,6 +1420,14 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
         } else
             upd = supd->ins->upd;
 
+        /*
+         * For in-memory WiredTiger configurations, reconciled prepare updates may require a
+         * tombstone when they are rolled back. Set WT_UPDATE_PREPARE_RESTORED_FROM_DS to indicate
+         * this.
+         */
+        if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY) && upd->prepare_state == WT_PREPARE_INPROGRESS)
```

Comment:
This is not what I mean. This flag is already set when you try to rollback the prepared update. If you see this flag and it is in memory case, we should add a tombstone. No need to do anything here.

Reply:
Removed in 56db67b